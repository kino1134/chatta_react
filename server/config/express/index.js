import express from 'express'
// import session from 'express-session'
// import connectRedis from 'connect-redis'

import helmet from 'helmet'
import cors from 'cors'
// import compression from 'compression'
import morgan from 'morgan'
import bodyParser from 'body-parser'
import handler from './handler'
import { env, api } from '../'

export default (root, routes) => {
  const app = express()

  if (env === 'production' || env === 'development') {
    app.use(helmet())
    app.use(cors({ origin: api.corsOrigin }))
    // 圧縮は必要性が感じられるまでコメントアウト
    // app.use(compression())
    app.use(morgan('dev'))
  }

  // TODO: Cookieのドメインが違う場合にチェックが通らなくなるので、一旦使わない
  // ソーシャルログイン時、stateチェックを行うために使用
  // const RedisStore = connectRedis(session)
  // const store = new RedisStore({
  //   host: redis.host,
  //   port: redis.port,
  //   pass: redis.password
  // })
  // app.use(session({
  //   secret: api.sessionSecret,
  //   resave: false,
  //   saveUninitialized: false,
  //   store }))

  app.use(bodyParser.urlencoded({ extended: false }))
  app.use(bodyParser.json())
  app.use(root, routes)
  app.use(handler.routeNotFound)
  app.use(handler.error(env))

  return app
}
