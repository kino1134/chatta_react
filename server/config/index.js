import dotenv from 'dotenv'

const env = process.env.NODE_ENV || 'development'

// Load .env per NODE_ENV
dotenv.config({ path: '.env.' + env })

// Load .env
const result = dotenv.config()
if (result.error) {
  throw result.error
}

// 環境変数の読み込み（必須チェック付き）
const required = (name) => {
  if (!process.env[name]) {
    throw new Error('You must set the ' + name + ' environment variable')
  }
  return process.env[name]
}

const config = {
  env: env,
  api: {
    port: process.env.API_PORT || 3003,
    ip: process.env.API_IP || '0.0.0.0',
    root: process.env.API_ROOT || '/api',
    sessionSecret: process.env.API_SESSION_SECRET || 'keyboard cat',
    corsOrigin: (process.env.API_CORS_ORIGIN || 'http://localhost:3000 http://localhost').split(' ')
  },
  client: {
    uri: process.env.CLIENT_URI || 'http://localhost:3000'
  },
  socket: {
    port: process.env.SOCKET_PORT
  },
  mongo: {
    uri: required('MONGO_URI'),
    user: required('MONGO_USER'),
    password: required('MONGO_PASSWORD'),
    debug: process.env.MONGO_DEBUG || true
  },
  redis: {
    host: process.env.REDIS_HOST || 'localhost',
    port: process.env.REDIS_PORT || 6379,
    password: process.env.REDIS_PASSWORD
  },
  jwt: {
    secret: required('JWT_SECRET_KEY'),
    expire: required('JWT_EXPIRES_IN')
  },
  google: {
    clientId: process.env.GOOGLE_CLIENT_ID || 'dummy',
    clientSecret: process.env.GOOGLE_CLIENT_SECRET || 'dummy'
  },
  github: {
    clientId: process.env.GITHUB_CLIENT_ID || 'dummy',
    clientSecret: process.env.GITHUB_CLIENT_SECRET || 'dummy'
  },
  sendgrid: {
    enabled: process.env.SENDGRID_ENABLED || 'true',
    apiKey: process.env.SENDGRID_API_KEY || 'dummy'
  }
}

module.exports = config
export default module.exports
