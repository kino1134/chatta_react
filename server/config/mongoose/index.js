import mongoose from 'mongoose'
import { mongo } from '../'

mongoose.set('debug', mongo.debug)

// ObjectIdの表示用メソッド
mongoose.Types.ObjectId.prototype.view = function () {
  return { id: this.toString() }
}

mongoose.connection.on('error', (err) => {
  console.error('MongoDB connection error:' + err)
  process.exit(-1)
})

export default mongoose
