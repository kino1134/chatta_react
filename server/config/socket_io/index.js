import { createClient } from 'redis'
import socketIo from 'socket.io'
import socketRedis from 'socket.io-redis'
import { redis } from '../'

export default () => {
  const io = socketIo()

  // scaling
  const pub = createClient({ host: redis.host, port: redis.port, password: redis.password })
  const sub = createClient({ host: redis.host, port: redis.port, password: redis.password })
  io.adapter(socketRedis({ pubClient: pub, subClient: sub }))

  // TODO: Test Code
  io.on('connection', function (socket) {
    console.log('connected socket')

    socket.on('disconnect', function () {
      console.log('disconnected socket')
    })
  })

  return io
}
