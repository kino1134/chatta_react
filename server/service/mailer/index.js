import mailer from '@sendgrid/mail'
import ejs from 'ejs'
import path from 'path'
import fs from 'fs'

import { sendgrid } from '../../config'

export default {
  send ({ to, subject, text, html }) {
    if (sendgrid.enabled === 'false') {
      console.log(text)
      console.log(html)
      return true
    }

    mailer.setApiKey(sendgrid.apiKey)

    const from = 'test@example.com'
    const message = {
      from, to, subject: (subject || '') + ' - chatta'
    }
    if (text) message.text = text
    if (html) message.html = html

    mailer.send(message)
  },

  render (fileName, data) {
    const filePath = path.join(process.cwd(), 'server/views', fileName)
    return ejs.compile(fs.readFileSync(filePath, 'utf-8'))(data)
  }
}
