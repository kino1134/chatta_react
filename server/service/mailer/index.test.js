import sg from '@sendgrid/mail'
import nock from 'nock'
import mailer from '.'

import { sendgrid } from '../../config'

let send
beforeEach(() => {
  nock('https://api.sendgrid.com')
    .filteringRequestBody(() => '*')
    .post('/v3/mail/send', '*')
    .reply(202)
  send = jest.spyOn(sg, 'send')
})

describe('render', () => {
  it('singUp', () => {
    const data = {
      user: {displayName: 'なまえ', userId: 1234},
      password: 'Password1'
    }
    expect(mailer.render('signUp.ejs', data)).toMatchSnapshot()
  })

  it('passwordInit', () => {
    const data = {
      user: {displayName: 'なまえ', userId: 1234},
      password: 'newPassword'
    }
    expect(mailer.render('passwordInit.ejs', data)).toMatchSnapshot()
  })
})

describe('send', () => {
  const data = {
    to: 'to@example.co.jp',
    subject: 'title',
    text: 'テキスト',
    html: '<p>paragraph</p>'
  }

  it('check bodyLog', async () => {
    const log = jest.spyOn(console, 'log')
    await mailer.send(data)
    expect(log).toBeCalledWith(data.text)
    expect(log).toBeCalledWith(data.html)
  })

  it('check content', async () => {
    sendgrid.enabled = true
    const res = await mailer.send(data)
    expect(res[0].statusCode).toBe(202)
    expect(send).toBeCalledWith(expect.objectContaining({
      to: data.to,
      text: data.text,
      html: data.html,
      subject: data.subject + ' - chatta'
    }))
  })
})
