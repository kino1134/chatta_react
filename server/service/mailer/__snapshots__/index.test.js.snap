// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render passwordInit 1`] = `
"なまえ さんのパスワードが初期化されました。<br/><br/>
ID：1234<br/>
新しいパスワード：newPassword
"
`;

exports[`render singUp 1`] = `
"なまえ さんのユーザが登録されました。<br/><br/>
ID：1234<br/>
パスワード：Password1
"
`;
